### PR TITLE
Enable golint in CI

### PR DESCRIPTION
**Pull request by SuperQ**
Enable the golint module in golangci-lint.

Fixes: https://github.com/prometheus/prometheus/issues/4125

Signed-off-by: Ben Kochie <superq@gmail.com>

### CODE REVIEW

**Comment by SuperQ** [comment]
There are two additional lint items I haven't fixed yet.

```
tsdb/db.go:42:2: a blank import should be only in a main or test package, or have a comment justifying it (golint)
	_ "github.com/prometheus/prometheus/tsdb/goversion"
	^
tsdb/wal/wal.go:840:44: exported func NewSegmentBufReader returns unexported type *wal.segmentBufReader, which can be annoying to use (golint)
func NewSegmentBufReader(segs ...*Segment) *segmentBufReader {
                                           ^
```

**Comment by SuperQ** [comment]
For the tsdb/goversion issue, I think a comment explaining that it's to load the package into main is justified.

**Comment by brian-brazil** [comment]
> For the tsdb/goversion issue, I think a comment explaining that it's to load the package into main is justified.

I don't see the need for a comment at all, this sounds more like a personal preference than a lint issue.

**Comment by mdlayher** [comment]
"Load the package into main" is implied by the import statement; what sort of initialization is happening with the underscore import? Leave a quick comment explaining that and move on. It's less hassle to not fight the linter and move on.

**Comment by beorn7** [comment]
Can't you just put a golint pragma there to disable the warning? We can later think about more invasive/controversial changes. It's still better to have two more "TODOs" in the code than missing all the other lint issues.

**Comment by SuperQ** [comment]
@beorn7 Yes, you can `nolint:modulename`. I wanted to fix as many things as many reasonable changes as possible before resorting to ignoring things.

**Comment by SuperQ** [comment]
All lint issues resolved.

File: pkg/textparse/openmetricsparse.go [hunk 1]
```diff
@@ -233,7 +233,7 @@ func (p *OpenMetricsParser) Next() (Entry, error) {
 	p.hasExemplarTs = false
 
 	switch t := p.nextToken(); t {
-	case tEofWord:
+	case tEOFWord:
```
Comment (beorn7) [inline pkg/textparse/openmetricsparse.go]: Shouldn't this be `EOfWord`?

File: tsdb/wal/wal.go [hunk 2]
```diff
@@ -837,6 +837,7 @@ type segmentBufReader struct {
 	off  int // Offset of read data into current segment.
 }
 
+// nolint:golint
```
Comment (beorn7) [inline tsdb/wal/wal.go]: Perhaps add something like `TODO: Consider exporting segmentBufReader`?

Comment (SuperQ) [reply tsdb/wal/wal.go]: Sure.

File: pkg/textparse/openmetricsparse.go (hunk 1 above)
Comment (mdlayher) [reply pkg/textparse/openmetricsparse.go]: Go precedent is `EOF`, as in `io.EOF`.

File: promql/parser/printer.go [hunk 3]
```diff
@@ -113,7 +113,7 @@ func (node *Call) String() string {
 
 func (node *MatrixSelector) String() string {
 	// Copy the Vector selector before changing the offset
-	var vecSelector VectorSelector = *node.VectorSelector.(*VectorSelector)
+	var vecSelector = *node.VectorSelector.(*VectorSelector)
```
Comment (mdlayher) [inline promql/parser/printer.go]: Nit: should just be `:=`.

File: pkg/textparse/openmetricsparse.go (hunk 1 above)
Comment (beorn7) [reply pkg/textparse/openmetricsparse.go]: `EOF` is an acronym that stands for "End Of File".

In `EOfWord` only the `E` is an abbreviation, it stands for "End Of Word".

`EOFWord` would suggest "End Of File Word" or "End Of Foo Word" or something.

Comment (beorn7) [reply pkg/textparse/openmetricsparse.go]: And now I have to correct myself, it actually does stand for "End Of File Word" or "EOF Word". So please forget my whole line of argument here. All good.

Comment (mdlayher) [reply pkg/textparse/openmetricsparse.go]: Ack, it didn't even occur to me that this might not mean "end-of-file". Apologies for jumping in!

Comment (beorn7) [reply pkg/textparse/openmetricsparse.go]: But it does. You were right, and I was wrong. Thanks for jumping in! :o)

File: promql/parser/printer.go (hunk 3 above)
Comment (SuperQ) [reply promql/parser/printer.go]: Ohh, good one.

File: pkg/labels/labels.go [hunk 4]
```diff
@@ -301,16 +301,14 @@ func Compare(a, b Labels) int {
 		if a[i].Name != b[i].Name {
 			if a[i].Name < b[i].Name {
 				return -1
-			} else {
-				return 1
 			}
+			return 1
```
Comment (roidelapluie) [inline pkg/labels/labels.go]: Should we be consistent with https://github.com/prometheus/prometheus/pull/6845

Comment (beorn7) [reply pkg/labels/labels.go]: I don't understand #6845. How can that be by design?

Comment (roidelapluie) [reply pkg/labels/labels.go]: This was left as it until now as it is easier to reason about. (that was @brian-brazil 's call at the time)

Comment (roidelapluie) [reply pkg/labels/labels.go]: But yeah, if we need to chose between this and addinfg a comment to disable the check, let's pick this solution.

**Pull request merged**